tests/components/azure_event_hub/conftest.py

```diff
@@ -0,0 +1,126 @@
+"""Test fixtures for AEH."""
+from dataclasses import dataclass
+from datetime import timedelta
+import logging
+from unittest.mock import MagicMock, patch
+
+from azure.eventhub.aio import EventHubProducerClient
+import pytest
+
+from homeassistant.components.azure_event_hub.const import (
+    CONF_FILTER,
+    CONF_SEND_INTERVAL,
+    DOMAIN,
+)
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import STATE_ON
+from homeassistant.setup import async_setup_component
+from homeassistant.util.dt import utcnow
+
+from .const import AZURE_EVENT_HUB_PATH, BASIC_OPTIONS, PRODUCER_PATH, SAS_CONFIG_FULL
+
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+_LOGGER = logging.getLogger(__name__)
+
+
+# fixtures for both init and config flow tests
+@pytest.fixture(autouse=True, name="mock_get_eventhub_properties")
+def mock_get_eventhub_properties_fixture():
+    """Mock azure event hub properties, used to test the connection."""
+    with patch(f"{PRODUCER_PATH}.get_eventhub_properties") as get_eventhub_properties:
+        yield get_eventhub_properties
+
+
+@pytest.fixture(name="filter_schema")
+def mock_filter_schema():
+    """Return an empty filter."""
+    yield {}
+
+
+@pytest.fixture(name="entry")
+async def mock_entry_fixture(hass, filter_schema, mock_create_batch, mock_send_batch):
+    """Create the setup in HA."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=SAS_CONFIG_FULL,
+        title="test-instance",
+        options=BASIC_OPTIONS,
+    )
+    entry.add_to_hass(hass)
+    assert await async_setup_component(
+        hass, DOMAIN, {DOMAIN: {CONF_FILTER: filter_schema}}
+    )
+    assert entry.state == ConfigEntryState.LOADED
+
+    # Clear the component_loaded event from the queue.
+    async_fire_time_changed(
+        hass,
+        utcnow() + timedelta(seconds=entry.options[CONF_SEND_INTERVAL]),
+    )
+    await hass.async_block_till_done()
+    yield entry
```

Comment:
We only need to yield when we're inside a context manager and want to keep the context. Otherwise we can return the value.